### PR TITLE
Endpoints

### PR DESCRIPTION
**Pull request by EricLarson2020**
Created endpoints for covid, names, and dates

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+NYT_API_KEY: <%= ENV['NYT_API_KEY'] %>
```
Comment (jpc20) [inline config.yml:1]: Were you able to get this working?

File: app/services/nytimes_service.rb [hunk 2]
```diff
@@ -7,6 +7,34 @@ def get_obituaries
...
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_covid_obituaries
+    response = conn.get do |req|
+      req.params['fq'] = 'section_name:(obituaries)'
+      req.params['q'] = 'coronavirus'
+    end
+    JSON.parse(response.body, symbolize_names: true)
```
Comment (jpc20) [inline app/services/nytimes_service.rb:27]: Nice job setting these up!

File: Gemfile.lock [hunk 3]
```diff
@@ -15,19 +20,26 @@ GEM
...
     ruby2_keywords (0.0.2)
+    safe_yaml (1.0.5)
     sinatra (2.0.8.1)
       mustermann (~> 1.0)
       rack (~> 2.0)
       rack-protection (= 2.0.8.1)
       tilt (~> 2.0)
     tilt (2.0.10)
+    vcr (6.0.0)
+    webmock (3.8.3)
```
Comment (jpc20) [inline Gemfile.lock]: Do you think this is why the build is failing? Maybe it is not allowing requests cause these are still in here

File: app/test/obituary_test.rb [hunk 4]
```diff
@@ -15,7 +15,22 @@ def test_it_returns_a_200
...
+	end
+
+	def test_it_returns_formatted_obituary_data_by_name
+			get '/name', name: "George"
+			assert_equal "obituary", JSON.parse(last_response.body).first['data']['type']
+	end
+
+	def test_it_returns_formatted_obituary_data_by_date
+			get '/date', beginning_date: '20170101', ending_date: '20190101'
+			assert_equal "obituary", JSON.parse(last_response.body).first['data']['type']
```
Comment (jpc20) [inline app/test/obituary_test.rb:34]: Nice. Once we are set on the attributes we are returning, we can come back and make these tests more robust to be sure we are getting the data we want.

File: app/services/nytimes_service.rb [hunk 5]
```diff
@@ -7,6 +7,34 @@ def get_obituaries
...
+  end
+
+  def get_obituaries_by_date(beginning_date, ending_date)
+    response = conn.get do |req|
+      req.params['fq'] = "section_name:(obituaries)"
+      req.params['facet_fields'] = 'pub_year'
+      req.params['facet'] = 'true'
+      req.params['begin_date'] = "#{beginning_date}"
+      req.params['ending_date'] = "#{ending_date}"
+    end
```
Comment (stellakunzang) [inline app/services/nytimes_service.rb:37]: we might have to do some finagling of the date somewhere along the way. We are (probably) going to pass in a 4 digit year and in order to get all obits for the year the beginning and end date will need to be formatting yyyy0101 and yyyy1231. If we want to have search by specific date or month, we will just need to format the date accordingly for the NYT api to be happy.

File: app/test/obituary_test.rb (hunk 4 above)
Comment (stellakunzang) [reply app/test/obituary_test.rb:34]: definitely, looks great for now!

**Review by stellakunzang — APPROVED** [review]
This is a great start. I think we are about ready to make API calls from the main rails app and see what we are able to do with this!

File: Gemfile.lock (hunk 3 above)
Comment (jpc20) [reply Gemfile.lock]: Answer was yes, I just ran bundle install and everything is passing now.

**Review by jpc20 — APPROVED** [review]
Nice work Eric!

File: app/poros/nytimes_results.rb [hunk 6]
```diff
@@ -3,7 +3,28 @@
...
+
+  def get_obituaries_by_name(name)
+    json = NytimesService.new.get_obituaries_by_name(name)
+    json[:response][:docs].map do |obituary_data|
+      Obituary.new(format_obituary_data(obituary_data))
+    end
+  end
+
+  def get_obituaries_by_date(beginning_date, ending_date)
+    json = NytimesService.new.get_obituaries_by_date(beginning_date, ending_date)
```
Comment (sagemlee) [inline app/poros/nytimes_results.rb:27]: Looking good.

**Review by sagemlee — APPROVED** [review]

**Pull request merged**